### DISCUSSION TITLE
Methods whose name ends with "?" return false if the receiver value is nil or false

### DISCUSSION THREAD

**Discussion by ledsun** [ruby/ruby.wasm#434] [General]
Currently methods whose name ends with "?" return true only if the receiver is true.

Browser's [Element.closest](https://developer.mozilla.org/ja/docs/Web/API/Element/closest) method returns null when no matching element is found.
Using this API in ruby.wasm, we get the following.

```ruby
if event[:target].closest(selectors) != JS::Null
  # do something
end
```

If meethods whose name ends with "?" return false if the receiver value is nil or false, we get the following.

```ruby
if event[:target].closest?(selectors) 
  # do something
end
```

This would violate the existing ruby.wasm API design policy of not hiding JavaScript information as much as possible.
I think it would be easier for Rubyists to use if only nil and false were false as in Ruby.
What do you think?

If you have a good name, it would be good to create a new method that judges loosely.

**Comment by kateinoigakukun** [comment]
How about applying JavaScript's `Boolean(v)` against returned value when calling methods with `?`?

**Reply by ledsun** [reply]
Your idea is sufficient for my application.
I have implemented it in #443 to try it out.
